tests/behaviour/typeql/steps.rs

```diff
@@ -356,4 +358,120 @@ generic_step_impl! {
             matched entries of given {actual_answers}."
         );
     }
+
+    #[step(expr = "rules contain: {label}")]
+    async fn rules_contain(context: &mut Context, rule_label: LabelParam) {
+        let res = context.transaction().logic().get_rule(rule_label.name).await;
+        assert!(res.is_ok(), "{res:?}");
+    }
+
+    #[step(expr = "rules do not contain: {label}")]
+    async fn rules_do_not_contain(context: &mut Context, rule_label: LabelParam) {
+        let res = context.transaction().logic().get_rule(rule_label.name).await;
+        assert!(res.is_err(), "{res:?}");
+    }
+
+    #[step(expr = "rules are")]
+    async fn rules_are(context: &mut Context, step: &Step) {
+        let stream = context.transaction().logic().get_rules();
+        assert!(stream.is_ok(), "{:?}", stream.err());
+        let res = stream.unwrap().try_collect::<Vec<_>>().await;
+        assert!(res.is_ok(), "{:?}", res.err());
+        let answers = res.unwrap();
+        let step_table = iter_table_map(step).collect::<Vec<_>>();
+        let expected_answers = step_table.len();
+        let actual_answers = answers.len();
+        assert_eq!(
+            actual_answers, expected_answers,
+            "The number of identifier entries (rows) should match the number of answers, \
+            but found {expected_answers} identifier entries and {actual_answers} answers."
+        );
+        let mut matched_rows = 0;
+        for ans_row in &answers {
+            for table_row in &step_table {
+                if match_answer_rule(table_row, ans_row).await {
+                    matched_rows += 1;
+                    break;
+                }
+            }
+        }
+        assert_eq!(
+            matched_rows, actual_answers,
+            "An identifier entry (row) should match 1-to-1 to an answer, but there are only {matched_rows} \
+            matched entries of given {actual_answers}."
+        );
+    }
+
+    #[step(expr = "reasoning schema")]
```

Comment:
So as a consequence these steps only applicable to reasoner tests will move into their own file

Reply:
Now we can't use step functions from other files because `generic_step_impl!` macro doesn't understand `pub`.